core/src/main/java/org/apache/calcite/rel/metadata/RelMetadataQuery.java

```diff
@@ -145,9 +184,10 @@ private RelMetadataQuery(@SuppressWarnings("unused") boolean dummy) {
     this.lowerBoundCostHandler = initialHandler(BuiltInMetadata.LowerBoundCost.Handler.class);
   }
 
-  private RelMetadataQuery(JaninoRelMetadataProvider metadataProvider,
+  private RelMetadataQuery(
+      @Nullable MetadataHandlerProvider metadataHandlerProvider,
```

Comment:
Why is this nullable?

Reply:
Good catch. It was due to an earlier version of the refactor. Removed.